### ISSUE TITLE
stream/topic autocomplete

### ISSUE THREAD

**Issue by chrisbobbe** [zulip/zulip-flutter#124]
After #49, which is likely to inform the architecture here.

**Comment by ryanwwest** [comment]
I'm excited for this feature!

**Comment by gnprice** [comment]
For anyone picking up this issue: the PR #884 has previous work that might be helpful to borrow from. See Zulip's guide to continuing unfinished work:
https://zulip.readthedocs.io/en/latest/contributing/continuing-unfinished-work.html

**Comment by alya** [comment]
https://github.com/zulip/zulip/issues/36370 has a plan for updating channel ordering in the web app typeahead. We can have refining the ordering be a follow-up in the mobile app, but could maybe pull in some basic ideas, like handling deactivated and unsubscribed channels, if we aren't doing that already.

**Comment by sm-sayedi** [comment]
Yeah, that would be a good follow-up. Right now in #1902, we do follow the criteria in [`typeahead_helper.sort_streams`](https://github.com/zulip/zulip/blob/9fe24279d7208233d003d1313abc95f654decf4d/web/src/typeahead_helper.ts#L998-L1008) for sorting in autocomplete.
